base/src/main/java/org/aya/core/term/StructCall.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020-2022 Yinsen (Tesla) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.core.term;
+
+import kala.collection.immutable.ImmutableSeq;
+import kala.control.Option;
+import kala.tuple.Tuple2;
+import org.aya.concrete.stmt.ClassDecl;
+import org.aya.concrete.stmt.TeleDecl;
+import org.aya.core.def.FieldDef;
+import org.aya.core.def.StructDef;
+import org.aya.generic.Arg;
+import org.aya.ref.DefVar;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * @author zaoqi
+ */
+public record StructCall(
+  @NotNull DefVar<StructDef, ClassDecl.StructDecl> ref,
+  int ulift,
+  @NotNull ImmutableSeq<Tuple2<DefVar<FieldDef, TeleDecl.StructField>, Arg<@NotNull Term>>> params
+) implements Term {
```

Comment:
Some question:
- Why move out from `CallTerm`s? This class is pretty short and simple.
- Why name arguments `params` while other `CallTerm`s call them `args`?
- Original credits (the `@author` in Java doc)? Is this class a complete rewrite?

Reply:
Yes, I asked for a complete rewerite